src/test/java/com/meilisearch/integration/TenantTokenTest.java

```diff
@@ -0,0 +1,244 @@
+
+        assertEquals(0, searchResult.getHits().size());
+        assertEquals(20, searchResult.getLimit());
+    }
+
+    /** Test Create Tenant Token with search rules on one index */
+    @Test
+    public void testGenerateTenantTokenOnOneIndex() throws Exception {
+        String indexUid = "GenerateTokenOnOneIndex";
+        Index index = createEmptyIndex(indexUid);
+        Key key = getPrivateKey();
+
+        Client privateClient = new Client(new Config("http://localhost:7700", key.getKey()));
+
+        Map<String, Object> rules = new HashMap<String, Object>();
+        rules.put(indexUid, new HashMap<String, Object>());
+
+        String jwtToken = privateClient.generateTenantToken(rules);
+
+        Client tokenClient = new Client(new Config("http://localhost:7700", jwtToken));
+
+        SearchResult searchResult = tokenClient.index(indexUid).search("");
+
+        assertEquals(0, searchResult.getHits().size());
+        assertEquals(20, searchResult.getLimit());
+    }
+
+    /** Test Create Tenant Token with filter on search rules */
+    @Test
+    public void testGenerateTenantTokenWithFilter() throws Exception {
+        Key key = getPrivateKey();
+        String indexUid = "GenerateTokenwithFilter";
+
+        Client privateClient = new Client(new Config("http://localhost:7700", key.getKey()));
+
+        Map<String, Object> filters = new HashMap<String, Object>();
+        filters.put("filter", "id > 1");
+        Map<String, Object> rules = new HashMap<String, Object>();
+        rules.put("GenerateTokenwithFilter", filters);
+
+        String jwtToken = privateClient.generateTenantToken(rules);
+
+        Client tokenClient = new Client(new Config("http://localhost:7700", jwtToken));
+
+        Index index = client.index(indexUid);
+        TestData<Movie> testData = this.getTestData(MOVIES_INDEX, Movie.class);
+        Task task = index.addDocuments(testData.getRaw());
+        index.waitForTask(task.getUid());
+
+        Settings settings = index.getSettings();
+        settings.setFilterableAttributes(new String[] {"id"});
+        index.waitForTask(index.updateSettings(settings).getUid());
+
+        SearchResult searchResult = tokenClient.index(indexUid).search("");
+
+        assertEquals(20, searchResult.getHits().size());
+        assertEquals(20, searchResult.getLimit());
+        assertEquals(30, searchResult.getNbHits());
+    }
+
+    /** Test Create Tenant Token with expiration date */
+    @Test
+    public void testGenerateTenantTokenWithExpiresAt() throws Exception {
+        String indexUid = "GenerateTokenWithExpiresAt";
+        Index index = createEmptyIndex(indexUid);
+        Key key = getPrivateKey();
+
+        Client privateClient = new Client(new Config("http://localhost:7700", key.getKey()));
+
+        Map<String, Object> rules = new HashMap<String, Object>();
+        rules.put("*", new HashMap<String, Object>());
+
+        Instant instant = Instant.now();
```

Comment:
This is only supported by Java 8, right?

Reply:
Yes, it is the reason I just use it in the tests and not in the code. Do you think it could be a problem?

Reply:
I think the Instant is a great addition, but since we support java 7, we need to be able to run the test suite with java 7. We should work to create a process for this kind of situation. So I think the better is to remove the non-supported features in Java 7.

Reply:
Remove!